### PR TITLE
DROP USER statement and related updates

### LINKED ISSUE #1591: sql: Adding DROP USER command
PR: https://github.com/cockroachdb/cockroach/pull/16442

### PR DESCRIPTION
**Pull request by jseldess**
- Add docs on `DROP USER`
- Add `DROP USER` to sidenav and sql statements overview
- Update `CREATE USER` diagram
- Update related docs with guidance on removing users

Fixes #1591

### CODE REVIEW

File: v1.0/create-and-manage-users.md [hunk 1]
```diff
@@ -4,17 +4,18 @@ summary: To create and manage your cluster's users (which lets you control SQL-l
...
+## Considerations
 
 - Usernames are case-insensitive; must start with either a letter or underscore; must contain only letters, numbers, or underscores; and must be between 1 and 63 characters.
 - After creating users, you must [grant them privileges to databases and tables](grant.html).
 - On secure clusters, you must [create client certificates for users](create-security-certificates.html#create-the-certificate-and-key-pair-for-a-client) and users must [authenticate their access to the cluster](#user-authentication).
-
-{{site.data.alerts.callout_info}}You can also create users through the <a href="create-user.html"><code>CREATE USER</code></a> statement.{{site.data.alerts.end}}
-
-<div id="toc"></div>
+- Before or after removing users, you must [revoke thir privileges](revoke.html); otherwise, if users are created later with identicial usernames, they will inherit the old users' privileges.
```
Comment (mberhault) [inline v1.0/create-and-manage-users.md]: s/revoke thir privileges/revoke their privileges/

File: v1.1/create-user.md [hunk 2]
```diff
@@ -6,13 +6,16 @@ toc: false
...
+<div id="toc"></div>
+
+## Considerations
 
 - Usernames are case-insensitive; must start with either a letter or underscore; must contain only letters, numbers, or underscores; and must be between 1 and 63 characters.
 - After creating users, you must [grant them privileges to databases and tables](grant.html).
-- On secure clusters, users must [authenticate their access to the cluster](#user-authentication).
-
-{{site.data.alerts.callout_info}}You can also create and manage users through the <a href="create-and-manage-users.html"><code>cockroach user</code></a> command. Notably, this is the most efficient way to <a href="create-and-manage-users.html#remove-a-user">remove users</a>.{{site.data.alerts.end}}
+- On secure clusters, you must [create client certificates for users](create-security-certificates.html#create-the-certificate-and-key-pair-for-a-client) and users must [authenticate their access to the cluster](#user-authentication).
```
Comment (mberhault) [inline v1.1/create-user.md]: do we mention connecting with password only (no certificate) in secure mode?

Comment (jseldess) [reply v1.1/create-user.md]: Yes, we mention it briefly down in https://www.cockroachlabs.com/docs/dev/create-and-manage-users.html#user-authentication. I thought we had a docs issue to revise this content, but I can't find it. When you have a bit of time, would you mind opening an issue with feedback on what we need to explain better/more thoroughly?

File: v1.0/create-and-manage-users.md (hunk 1 above)
Comment (jseldess) [reply v1.0/create-and-manage-users.md]: Done.

**Pull request merged**